src/txkube/_compat.py

```diff
@@ -17,3 +17,13 @@ def dumps_bytes(obj):
     if isinstance(b, unicode):
         b = b.encode("ascii")
     return b
+
+
+
+def ensure_unicode(s, encoding="ascii", errors="strict"):
```

Comment:
Based on the changes so far and the uses that have been found to be incorrect, I suggest that this function should actually be named `native_string_to_bytes` and it should raise a `TypeError` when passed anything that is not a native  string.  This makes it completely safe to use to deal with the functions like `yaml.safe_load` and `json.dumps` which return native strings when we really want bytes and makes it difficult to use for cases like "well I can't remember/figure out what type of string this API produces so I'll just apply this hammer" (which I would prefer to avoid).